auth/basic/db.go

```diff
@@ -84,6 +101,32 @@ func (b *Store) CreateTx(ctx context.Context, tx *sql.Tx, userID, username strin
 	return err
 }
 
+// UpdateTx should update a user's password in auth_basic_users
+func (b *Store) UpdateTx(ctx context.Context, tx *sql.Tx, userID string, oldPassword ValidatedPassword, newPassword HashedPassword) error {
+	err := permission.LimitCheckAny(ctx, permission.Admin, permission.MatchUser(userID))
+	if err != nil {
+		return err
+	}
+
+	err = validate.UUID("UserID", userID)
+	if err != nil {
+		return err
+	}
+
+	if oldPassword != nil {
+		if userID != oldPassword.UserId() {
+			return validation.NewFieldError("oldPassword", "Password does not match User")
+		}
+	} else {
+		if !permission.Admin(ctx) {
+			return validation.NewFieldError("oldPassword", "Previous password required")
+		}
+	}
```

Comment:
looking back at this I have no idea why I nested the conditionals this way :/

Reply:
It's the struggle of re-using a single `oldPassword != nil` check in two conditionals but not having an elegant way to do both without nesting at least one.

I find myself doing the same thing occasionally in this situation. In the end, keeping it flatter and separate short-circuit returns with a bit of repetition vs. `else` tends to read better.

It may be better to swap the second one to `if !permission.Admin(ctx) && old == nil {` because `old != nil` and `old == nil` look very similar.
https://google.github.io/styleguide/go/guide#concision